src/main_nep/snes.cu

```diff
@@ -174,12 +249,14 @@ void SNES::regularize(Parameters& para)
   if (para.lambda_1 < 0.0f || para.lambda_2 < 0.0f) {
     float auto_reg = 1.0e30f;
     for (int p = 0; p < population_size; ++p) {
-      float temp = fitness[p + 3 * population_size] + fitness[p + 4 * population_size] +
-                   fitness[p + 5 * population_size];
+      float temp = fitness[p + (6 * para.num_types + 3) * population_size] +
+                   fitness[p + (6 * para.num_types + 4) * population_size] +
+                   fitness[p + (6 * para.num_types + 5) * population_size];
       if (auto_reg > temp) {
         auto_reg = temp;
       }
     }
+    auto_reg *= 0.4f;
```

Comment:
Not sure if 0.4 is good different systems. Let's do some more testings on this. (I'm running a couple of testings with different coefficients, which should be converged soon.)

Reply:
as we checked, 0.4 should be a good value.